trajopt_optimizers/trajopt_sqp/src/trust_region_sqp_solver.cpp

```diff
@@ -240,12 +241,34 @@ void TrustRegionSQPSolver::runTrustRegionLoop()
 
     // Solve the current QP problem
     status_ = solveQPProblem();
+
     if (status_ != SQPStatus::RUNNING)
     {
-      qp_problem->setVariables(results_.best_var_vals.data());
```

Comment:
This just needs to be added back.

Reply:
Done

Reply:
I don't think that would work since it would need to be set before convexify is called in stepSQPSolver. I think we might be able to move into that method but I would want to do a lot of testing to make sure everything aligns with the legacy version before making a change like this.

Reply:
Isn't this equivalent? In the previous version solveQPProblem() is called, and if qp_solver->solve() failed and hence solveQPProblem() failed setVariables is called from withing runTrustRegionLoop(). Now I moved this line to solveQPProblem(), still to be called if qp_solver->solve() failed. It's just clearer now, as solveQPProblem() either sets the variables to new_var_vals and back to best_var_vals if succeeded and immediately to best_var_vals if failed. See line 386 in my PR.

Reply:
Okay, It looks like we already handle setting back the values in solveQPProblem as you pointed out so it should not be needed. I would just recommend running a one of the test which is in both trajopt and trajopt ifopt that does not have a collision cost/constraint to verify that the output tables are the same. I mostly look at the bottom sum values for each iteration.

Reply:
Sure, can you point me to which ones?

Reply:
I would start with the numerical IK and maybe the simple collision unit test because it is not using the weighted collision approach.

Reply:
I'm still at this. There are differences, but not caused by this PR. I'm trying to find the root cause.

Reply:
I know the numerical ik has difference on Jammy because on one interaction it returns iteration limit instead of approximate solution but focal produces identical results. I am currently running focal so let me know if you want me to compare since I think you are running jammy.

Reply:
Aha, I'm running in that issue exactly! But I've also access to focal (I'm using WSL), so I'll run it there tomorrow.

Reply:
From what I could tell both hit the iteration limit but because of slight numerical differences focal picks approximate solution and jammy pick iteration limit.

Reply:
I will take a look